core-ui/src/components/ApiRules/ApiRulesList.js

```diff
@@ -11,7 +11,7 @@ export function ApiRulesList({ serviceName, namespace, prefix }) {
     isCompact: true,
     showTitle: true,
     createFormProps: { serviceName, prefix },
-    filter: apiRule => apiRule.spec.service.name === serviceName,
+    filter: apiRule => apiRule?.spec?.service?.name === serviceName,
```

Comment:
All those fields are mandatory when creating an APIRule, so imo this fix isn't  necessary.

Reply:
And still, it gets an error: `Cannot read properties of undefined (reading 'name')` 
<img width="887" alt="Screen Shot 2022-11-15 at 15 34 03" src="https://user-images.githubusercontent.com/20790348/201947215-ed3ae47c-cb2e-4337-878d-d9ca465f5fcc.png">

Reply:
It may receive empty apiRule so I removed the rest of the fixes